### PR TITLE
add facility to patient query

### PR DESCRIPTION
**Pull request by timothybest-usds**
## Related Issue or Background Info

resolves https://github.com/CDCgov/prime-simplereport/issues/452#issuecomment-752712747

## Changes Proposed

- add facility to the patient query
- add test for when the facility exists and when it is null

### CODE REVIEW

File: backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientDataResolver.java [hunk 1]
```diff
@@ -27,4 +29,9 @@
     public TestEvent getLastTest(Person p) {
         return _testEventService.getLastTestResultsForPatient(p);
     }
+
+    public ApiFacility getFacility(Person p) {
```
Comment (dmethvin-gov) [inline backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientDataResolver.java:33]: In this case I have to separately retrieve the entire list of facilities to populate the dropdown so all I need from the patient is the facility id. I suppose I can just grab the id like your test did. Not really sure if there are performance implications for doing it one way or the other.

**Review by dmethvin-gov — APPROVED** [review]

File: backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientDataResolver.java (hunk 1 above)
Comment (benwarfield-usds) [reply backend/src/main/java/gov/cdc/usds/simplereport/api/patient/PatientDataResolver.java:33]: There are notional performance implications,￼but for one patient nothing will matter all that much￼.

**Review by benwarfield-usds — CHANGES_REQUESTED** [review]
I believe there is a problem that should be fixed--if I'm weirdly wrong, please let me know!

File: backend/src/test/java/gov/cdc/usds/simplereport/api/PatientManagementTest.java [hunk 2]
```diff
@@ -7,21 +7,40 @@
...
 
+    @Autowired
+    private TestDataFactory _dataFactory;
+
     @BeforeEach
     public void setup() {
         truncateDb();
     }
 
+    public void queryPatientWithFacility() throws Exception {
```
[the remark this answers is not in the document]
Comment (timothybest-usds) [reply backend/src/test/java/gov/cdc/usds/simplereport/api/PatientManagementTest.java:39]: I thought it was odd all the test passed the first time I tried 🙃

File: backend/src/test/java/gov/cdc/usds/simplereport/api/PatientManagementTest.java [hunk 3]
```diff
@@ -7,21 +7,44 @@
...
     }
 
+    @Test
+    public void queryPatientWithFacility() throws Exception {
+        Organization org = _orgService.getCurrentOrganization();
+        Facility place = _orgService.getFacilities(org).get(0);
+        _dataFactory.createMinimalPerson(org, place, "Cassandra", null, "Thom", null);
+        _dataFactory.createMinimalPerson(org, null, " Miriana", "Linas", "Luisito", null);
+        JsonNode patients = fetchPatientsWithFacility();
+        assertEquals(true, patients.get(0).get("facility").isNull());
```
Comment (benwarfield-usds) [inline backend/src/test/java/gov/cdc/usds/simplereport/api/PatientManagementTest.java:44]: This can also just be `assertTrue`, but that method literally just delegates to this one. 

There's also a `path` method that lets you traverse possibly empty paths without getting exceptions until you check, but I don't think you need it here, that's just FYI. 🙂

**Review by benwarfield-usds — APPROVED** [review]
